[PL-50559] Adding note for OIDC connector

saifeemustafaqharness opened the pull request:
[PL-50559](https://harness.atlassian.net/browse/PL-50559)

This doc change is in reference to the request made on [PL-49120](https://harness.atlassian.net/browse/PL-49120)

Note added:

:::info note
OIDC authentication support is not available for any connectors in Legacy Delegates.

The support for OIDC is made available and tested from Delegate versions 24.03.82600 and higher.
:::

## Description

* Please describe your changes: \__________________________________
* Jira/GitHub Issue numbers (if any): \______________________________
* Preview links/images (Internal contributors only): \__________________

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

PRs must meet these requirements to be merged:

- [ ] Successful preview build.
- [ ] Code owner review.
- [ ] No merge conflicts.
- [ ] Release notes/new features docs: Feature/version released to at least one prod environment.


[PL-50559]: https://harness.atlassian.net/browse/PL-50559?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PL-49120]: https://harness.atlassian.net/browse/PL-49120?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

brian-f-harness commented on docs/first-gen/firstgen-platform/account/manage-connectors/add-kubernetes-cluster-cloud-provider.md:
```diff
@@ -270,6 +270,12 @@ Specify the encryption algorithm used when the certificate was created. Typicall
 These settings come from the OIDC provider authorization server you have set up and others come from the provider app you are using to log in with.
 
+:::info note
+OIDC authentication support is not available for any connectors in Legacy Delegates.
```
```suggestion
OIDC authentication support is not available for connectors in legacy delegates. For information on delegate types, go to [Delegate image types](/docs/platform/delegates/delegate-concepts/delegate-image-types).
```

brian-f-harness commented on docs/first-gen/firstgen-platform/account/manage-connectors/add-kubernetes-cluster-cloud-provider.md:
```diff
@@ -270,6 +270,12 @@ Specify the encryption algorithm used when the certificate was created. Typicall
+:::info note
+OIDC authentication support is not available for any connectors in Legacy Delegates.
+
+The support for OIDC is made available and tested from Delegate versions 24.03.82600 and higher.
```
```suggestion
OIDC support requires Harness Delegate version 24.03.82603 or later.
```

brian-f-harness commented on docs/first-gen/firstgen-platform/account/manage-connectors/add-kubernetes-cluster-cloud-provider.md:
```diff
@@ -270,6 +270,12 @@ Specify the encryption algorithm used when the certificate was created. Typicall
+OIDC authentication support is not available for any connectors in Legacy Delegates.
+
+The support for OIDC is made available and tested from Delegate versions 24.03.82600 and higher.
+:::
```
24.03.82603 was the released delegate image, so I think 24.03.82603 or later makes sense here

bot-gitexp-user commented:
Please check the Execution Link of the Pipeline for the Website Draft URL. This is located in the Preview Step behind the Harness VPN and also is available in #hdh_alerts. E.g Website Draft URL: https://unique-id--harness-developer.netlify.app. Current Draft URL is: https://6630f3657421b10ae03dfe73--harness-developer.netlify.app

brian-f-harness commented on docs/platform/connectors/cloud-providers/add-aws-connector.md:
```diff
@@ -58,6 +58,12 @@ Review the following information carefully to ensure you select the correct cred
    * **Use OIDC**: Select this option to connect to AWS with OIDC. This requires the user to create an OIDC identity provider in AWS and add it in a trust relationship with an IAM role that Harness will assume when doing any AWS operations with this connector.
 
+   :::info note
+   OIDC authentication support is not available for any connectors in Legacy Delegates.
```
```suggestion
   OIDC authentication support is not available for connectors in legacy delegates. For information on delegate types, go to [Delegate image types](/docs/platform/delegates/delegate-concepts/delegate-image-types).
```

brian-f-harness commented on docs/platform/connectors/cloud-providers/add-aws-connector.md:
```diff
@@ -58,6 +58,12 @@ Review the following information carefully to ensure you select the correct cred
+   :::info note
+   OIDC authentication support is not available for any connectors in Legacy Delegates.
+
+   The support for OIDC is made available and tested from Delegate versions 24.03.82600 and higher.
```
```suggestion
   OIDC support requires Harness Delegate version 24.03.82603 or later.
```

brian-f-harness reviewed (commented):
Note when referring to delegates in doc: https://harness.atlassian.net/wiki/spaces/DOCS/pages/21130772574/Documentation+Style+Guide#Platform

brian-f-harness commented on docs/platform/connectors/cloud-providers/connect-to-google-cloud-platform-gcp.md:
```diff
@@ -121,6 +121,11 @@ Before creating the GCP connector, you need to:
+   :::info note
+   OIDC authentication support is not available for any connectors in Legacy Delegates.
+
+   The support for OIDC is made available and tested from Delegate versions 24.03.82600 and higher.
```
```suggestion
   OIDC support requires Harness Delegate version 24.03.82603 or later.
```

brian-f-harness commented on docs/platform/connectors/cloud-providers/ref-cloud-providers/aws-connector-settings-reference.md:
```diff
@@ -435,6 +435,12 @@ Specify the credentials that enable Harness to connect your AWS account. There a
+   :::info note
+   OIDC authentication support is not available for any connectors in Legacy Delegates.
+
+   The support for OIDC is made available and tested from Delegate versions 24.03.82600 and higher.
```
```suggestion
   OIDC support requires Harness Delegate version 24.03.82603 or later.
```

brian-f-harness commented on docs/platform/connectors/cloud-providers/ref-cloud-providers/gcs-connector-settings-reference.md:
```diff
@@ -116,6 +116,12 @@ Select this option to allow the connector to inherit its authentication credenti
+:::info note
+OIDC authentication support is not available for any connectors in Legacy Delegates.
+
+The support for OIDC is made available and tested from Delegate versions 24.03.82600 and higher.
```
```suggestion
OIDC support requires Harness Delegate version 24.03.82603 or later.
```

brian-f-harness commented on docs/platform/connectors/ticketing-systems/connect-to-service-now.md:
```diff
@@ -100,7 +100,9 @@ To use ADFS credentials for authentication:
-This feature requires Harness Delegate 799xx or higher.
+OIDC authentication support is not available for any connectors in Legacy Delegates.
+
+The support for OIDC is made available and tested from Delegate versions 24.03.82600 and higher.
```
```suggestion
OIDC support requires Harness Delegate version 24.03.82603 or later.
```

brian-f-harness commented on docs/platform/connectors/cloud-providers/ref-cloud-providers/gcs-connector-settings-reference.md:
```diff
@@ -116,6 +116,12 @@ Select this option to allow the connector to inherit its authentication credenti
 Select this option if you want to [use OIDC](#use-openid-connect-oidc).
 
+:::info note
+OIDC authentication support is not available for any connectors in Legacy Delegates.
```
```suggestion
OIDC authentication support is not available for connectors in legacy delegates. For information on delegate types, go to [Delegate image types](/docs/platform/delegates/delegate-concepts/delegate-image-types).
```

brian-f-harness commented on docs/platform/connectors/cloud-providers/ref-cloud-providers/kubernetes-cluster-connector-settings-reference.md:
```diff
@@ -351,6 +351,12 @@ For more details, go to [Managing Service Accounts](https://kubernetes.io/docs/r
 These settings come from your OIDC provider authorization server or the provider app you use to log in.
 
+:::info note
+OIDC authentication support is not available for any connectors in Legacy Delegates.
```
```suggestion
OIDC authentication support is not available for connectors in legacy delegates. For information on delegate types, go to [Delegate image types](/docs/platform/delegates/delegate-concepts/delegate-image-types).
```

brian-f-harness commented on docs/platform/connectors/cloud-providers/ref-cloud-providers/kubernetes-cluster-connector-settings-reference.md:
```diff
@@ -351,6 +351,12 @@ For more details, go to [Managing Service Accounts](https://kubernetes.io/docs/r
+:::info note
+OIDC authentication support is not available for any connectors in Legacy Delegates.
+
+The support for OIDC is made available and tested from Delegate versions 24.03.82600 and higher.
```
```suggestion
OIDC support requires Harness Delegate version 24.03.82603 or later.
```

brian-f-harness reviewed (commented):
@saifeemustafaqharness This note should be a partial file. Let me know if I can help.

https://harness.atlassian.net/wiki/spaces/DOCS/pages/21327872338/Tips+hints+and+best+practices#Reuse-content-with-partial-files

bot-gitexp-user commented:
Please check the Execution Link of the Pipeline for the Website Draft URL. This is located in the Preview Step behind the Harness VPN and also is available in #hdh_alerts. E.g Website Draft URL: https://unique-id--harness-developer.netlify.app. Current Draft URL is: https://6631040db9203f1b6aedfda4--harness-developer.netlify.app

brian-f-harness commented on docs/platform/connectors/ticketing-systems/connect-to-service-now.md:
```diff
@@ -100,9 +100,9 @@ To use ADFS credentials for authentication:
 
 
```
@saifeemustafaqharness I don't think we need this info in this topic.

aimurphy replied:
It's not clear which version is needed. If it requires delegate version 24.03.82603 or later, why did it originally say 799xx or later?

brian-f-harness commented on docs/platform/connectors/cloud-providers/ref-cloud-providers/kubernetes-cluster-connector-settings-reference.md:
```diff
@@ -352,9 +352,9 @@ For more details, go to [Managing Service Accounts](https://kubernetes.io/docs/r
 These settings come from your OIDC provider authorization server or the provider app you use to log in.
 
```
@saifeemustafaqharness I don't think we need this info in this topic.

aimurphy replied:
I think this info could just be appended briefly to line 352. Please remove here and I will handle it in a separate PR.

brian-f-harness commented on docs/first-gen/firstgen-platform/account/manage-connectors/add-kubernetes-cluster-cloud-provider.md:
```diff
@@ -271,9 +271,9 @@ Specify the encryption algorithm used when the certificate was created. Typicall
 These settings come from the OIDC provider authorization server you have set up and others come from the provider app you are using to log in with.
 
```
@saifeemustafaqharness I don't think we need this info in this topic.

aimurphy replied:
Agree with @brian-f-harness. I wouldn't add this to first gen.

aimurphy commented on docs/platform/connectors/cloud-providers/ref-cloud-providers/gcs-connector-settings-reference.md:
```diff
@@ -116,6 +116,12 @@ Select this option to allow the connector to inherit its authentication credenti
+
+OIDC support requires Harness Delegate version 24.03.82603 or later.
+:::
+
```
This information is more relevant on line 156. Please revert/remove from this PR. I will handle in a separate PR.

aimurphy commented on docs/platform/connectors/cloud-providers/ref-cloud-providers/aws-connector-settings-reference.md:
```diff
@@ -435,6 +435,12 @@ Specify the credentials that enable Harness to connect your AWS account. There a
 
    :::
 
+   :::info note
```
This topic has too many notes in a row. I think this can be combined with the preceding note. Please remove here. I will handle on a separate PR.

aimurphy commented on docs/platform/connectors/cloud-providers/connect-to-google-cloud-platform-gcp.md:
```diff
@@ -121,6 +121,11 @@ Before creating the GCP connector, you need to:
    * **Use the credentials of a specific Harness Delegate:** Allow the connector to inherit its authentication credentials from the Harness Delegate that is running in GCP.
    * **OIDC:** Enables OIDC authentication. For more information, go to [Use OIDC](./ref-cloud-providers/gcs-connector-settings-reference.md#use-openid-connect-oidc).
+   :::info note
+   OIDC authentication support is not available for connectors in legacy delegates. For information on delegate types, go to [Delegate image types](/docs/platform/delegates/delegate-concepts/delegate-image-types).
```
This is not needed here. This is a generic topic. The details are given in the settings reference topic. Please remove.

aimurphy commented on docs/platform/connectors/cloud-providers/add-aws-connector.md:
```diff
@@ -58,6 +58,12 @@ Review the following information carefully to ensure you select the correct cred
    * **Use OIDC**: Select this option to connect to AWS with OIDC. This requires the user to create an OIDC identity provider in AWS and add it in a trust relationship with an IAM role that Harness will assume when doing any AWS operations with this connector.
 
+   :::info note
+   OIDC authentication support is not available for connectors in legacy delegates. For information on delegate types, go to [Delegate image types](/docs/platform/delegates/delegate-concepts/delegate-image-types).
```
This is not needed here. This is a generic topic. The details are given in the settings reference topic. I will update this topic so it's not repetitive of the reference topic. Please remove.

aimurphy commented:
Handled under separate PR. You can close this.

saifeemustafaqharness commented:
I have removed this from all irrelevant locations as suggested by April and Brian. Please review and let me know if there are any other places where this could be added. Currently, out of a total of 7 files containing OIDC references, I have removed it from 6, leaving it present only in the GCP connector document. We also need to include a mention of this where the AWS Connector is referenced for OIDC authentication.

@brian-f-harness

bot-gitexp-user commented:
Please check the Execution Link of the Pipeline for the Website Draft URL. This is located in the Preview Step behind the Harness VPN and also is available in #hdh_alerts. E.g Website Draft URL: https://unique-id--harness-developer.netlify.app. Current Draft URL is: https://66335a8bc28fae9ed3fba652--harness-developer.netlify.app

bot-gitexp-user commented:
Please check the Execution Link of the Pipeline for the Website Draft URL. This is located in the Preview Step behind the Harness VPN and also is available in #hdh_alerts. E.g Website Draft URL: https://unique-id--harness-developer.netlify.app. Current Draft URL is: https://66335ad29a91dea27b09f376--harness-developer.netlify.app

bot-gitexp-user commented:
Please check the Execution Link of the Pipeline for the Website Draft URL. This is located in the Preview Step behind the Harness VPN and also is available in #hdh_alerts. E.g Website Draft URL: https://unique-id--harness-developer.netlify.app. Current Draft URL is: https://66335ae654e7a19ffc3bcb80--harness-developer.netlify.app

brian-f-harness commented:
@saifeemustafaqharness This PR can be closed. Doc was added here - https://github.com/harness/developer-hub/pull/6526
